src/main/java/com/redhat/rhjmc/containerjfr/commands/internal/StartRecordingCommand.java

```diff
@@ -135,33 +135,34 @@ public void execute(String[] args) throws Exception {
     }
 
     @Override
-    public boolean validate(String[] args) {
+    public void validate(String[] args) throws FailedValidationException {
         if (args.length != 3) {
-            cw.println(
-                    "Expected three arguments: target (host:port, ip:port, or JMX service URL), recording name, and event types");
-            return false;
+            String errorMessage =
+                    "Expected three arguments: target (host:port, ip:port, or JMX service URL), recording name, and event types";
+            cw.println(errorMessage);
+            throw new FailedValidationException(errorMessage);
         }
 
         String targetId = args[0];
         String name = args[1];
         String events = args[2];
 
-        boolean isValidTargetId = validateTargetId(targetId);
-        boolean isValidName = validateRecordingName(name);
-        boolean isValidEvents = validateEvents(events);
-
-        if (!isValidTargetId) {
-            cw.println(String.format("%s is an invalid connection specifier", args[0]));
+        if (!validateTargetId(targetId)) {
+            String errorMessage = "%s is an invalid connection specifier";
```

Comment:
In this kind of scenario where there are multiple validation checks that can fail, beyond the most basic "wrong number of arguments" failure, how about performing all of the checks and constructing a message informing the client about all of the failures, rather than just the "first" in the arbitrary implementation order?

Reply:
This would match more closely with the previous behaviour, which would print out all of the failures before returning the value of the validation result.

Reply:
Ah, I didn't realize that.

For combining them, what do you think about just concatenating them together with some sort of separator? For example:
```
Multiple validation checks failed: %s is an invalid connection specifier; %s is an invalid recording name; %s is an invalid events specifier
```

And do you have any thoughts on adding tests to support this? I feel like being completely exhaustive and testing all the possible multiple-failure combinations is probably a bit too much? So maybe we could just do a 2-fail and all-fail test for `validate()` methods with more than two checks, and obviously just the 2-fail for any that have two checks?

Reply:
Yea, semicolons to separate them seems good to me.

I think it's probably worth exhaustively testing all the validation cases, even if it does seem repetitive/monotonous to do. It's easy for a refactor or some other change in the future to accidentally change the semantics of the method, and that's a regression case that unit tests are great for catching. The command validation step should be considered as part of our command channel API, so we want to be very sure that it's stable between versions.